dozer-sql/src/pipeline/builder.rs

```diff
@@ -26,6 +26,7 @@ pub struct SchemaSQLContext {}
 #[derive(Debug, Clone, Default)]
 pub struct QueryContext {
     pub cte_names: HashSet<String>,
+    pub pipeline_map: HashMap<String, (String, PortHandle)>,
 }
```

Comment:
You can leverage on pipeline_map to check the cte duplicates, every time a CTE is found a subpipeline is created and the name is the key of the hashmap

Reply:
done